pkgs/by-name/ve/vesktop/package.nix

```diff
@@ -0,0 +1,91 @@
+{ lib
+, fetchFromGitHub
+, substituteAll
+, buildNpmPackage
+, makeDesktopItem
+, copyDesktopItems
+, vencord
+, electron
+, pipewire
+, libicns
+}:
+let
+  libPath = lib.makeLibraryPath [ pipewire ];
+in
+buildNpmPackage rec {
+  pname = "vesktop";
+  version = "0.3.3";
+
+  src = fetchFromGitHub {
+    owner = "Vencord";
+    repo = "Vesktop";
+    rev = "v${version}";
+    sha256 = "sha256-Njs3tACxUyRolYUtS/q2lITIQnUBFXVXWZEfQ66HpPM=";
+  };
+
+  nativeBuildInputs = [ copyDesktopItems ];
+
+  prePatch = ''
+    cat ${./package-lock.json} > ./package-lock.json
+  '';
+
+  patches = [
+    (substituteAll { inherit vencord; src = ./use_system_vencord.patch; })
+  ];
+
+  npmDepsHash = "sha256-X4/7N+l92CfTfD3XXifRd5G0oxGWvIUl8CG+foAyShc=";
+  npmFlags = [ "--legacy-peer-deps" ];
+  makeCacheWritable = true;
+  env.ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
+
+  postBuild = ''
+    npm exec electron-builder -- \
+      --dir \
+      -c.electronDist=${electron}/libexec/electron \
+      -c.electronVersion=${electron.version}
+  '';
+
+  # this is consistent with other nixpkgs electron packages and upstream, as far as I am aware
+  # yes, upstream really packages it as "vesktop" but uses "vencorddesktop" file names
+  installPhase = ''
```

Comment:
i think it would be better to use pnpm directly here over maintaining our own lockfile. this implementation comes from a different package - [pot](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/pot/default.nix) - which allows for us to use the upstream pnpm lockfile and only update the hash on our end, similar to the current usage of `buildNpmPackage`. the only problem with this currently is that `ELECTRON_SKIP_BINARY_DOWNLOAD` doesn't seem to be respected 🤔 

```suggestion
{ lib
, stdenv
, stdenvNoCC
, fetchFromGitHub
, substituteAll
, makeDesktopItem
, copyDesktopItems
, vencord
, electron
, pipewire
, libicns
, jq
, moreutils
, nodePackages
}:
stdenv.mkDerivation rec {
  pname = "vesktop";
  version = "0.3.3";

  src = fetchFromGitHub {
    owner = "Vencord";
    repo = "Vesktop";
    rev = "v${version}";
    sha256 = "sha256-Njs3tACxUyRolYUtS/q2lITIQnUBFXVXWZEfQ66HpPM=";
  };

  pnpm-deps = stdenvNoCC.mkDerivation {
    pname = "${pname}-pnpm-deps";
    inherit src version patches ELECTRON_SKIP_BINARY_DOWNLOAD;

    nativeBuildInputs = [
      jq
      moreutils
      nodePackages.pnpm
    ];

    # https://github.com/NixOS/nixpkgs/blob/763e59ffedb5c25774387bf99bc725df5df82d10/pkgs/applications/misc/pot/default.nix#L56
    installPhase = ''
      export HOME=$(mktemp -d)

      pnpm config set store-dir $out
      pnpm install --frozen-lockfile --ignore-script

      rm -rf $out/v3/tmp
      for f in $(find $out -name "*.json"); do
        sed -i -E -e 's/"checkedAt":[0-9]+,//g' $f
        jq --sort-keys . $f | sponge $f
      done
    '';

    dontFixup = true;
    outputHashMode = "recursive";
    outputHash = "sha256-vInaSLGahRUgvwAeUcI+oV84L+tgNRCmfFpE0aUD4X4=";
  };

  nativeBuildInputs = [
    copyDesktopItems
    nodePackages.pnpm
    nodePackages.nodejs
  ];

  patches = [
    (substituteAll { inherit vencord; src = ./use_system_vencord.patch; })
  ];

  ELECTRON_SKIP_BINARY_DOWNLOAD = 1;

  preBuild = ''
    export HOME=$(mktemp -d)
    export STORE_PATH=$(mktemp -d)

    cp -r ${pnpm-deps}/* "$STORE_PATH"
    chmod -R +w "$STORE_PATH"

    pnpm config set store-dir "$STORE_PATH"
    pnpm install --offline --frozen-lockfile --ignore-script
    patchShebangs node_modules/{*,.*}
  '';

  postBuild = ''
    pnpm build
    pnpm exec electron-builder -- \
      --dir \
      -c.electronDist=${electron}/libexec/electron \
      -c.electronVersion=${electron.version}
  '';

  # this is consistent with other nixpkgs electron packages and upstream, as far as I am aware
  # yes, upstream really packages it as "vesktop" but uses "vencorddesktop" file names
  installPhase = let
    libPath = lib.makeLibraryPath [ pipewire ];
  in ''
```